cli/cmd/check.go

```diff
@@ -44,33 +36,7 @@ local system, the Linkerd control plane, and connectivity between those. The pro
 problems were found.`,
 		Args: cobra.NoArgs,
 		Run: func(cmd *cobra.Command, args []string) {
-
-			kubeApi, err := k8s.NewAPI(kubeconfigPath)
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "Error with Kubernetes API: %s\n", err.Error())
-				statusCheckResultWasError(os.Stdout)
-				os.Exit(2)
-			}
-
-			var apiClient pb.ApiClient
-			if apiAddr != "" {
-				apiClient, err = public.NewInternalClient(controlPlaneNamespace, apiAddr)
-			} else {
-				apiClient, err = public.NewExternalClient(controlPlaneNamespace, kubeApi)
-			}
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "Error with Linkerd API: %s\n", err.Error())
-				statusCheckResultWasError(os.Stdout)
-				os.Exit(2)
-			}
-
-			grpcStatusChecker := healthcheck.NewGrpcStatusChecker(apiClient)
-			versionStatusChecker := version.NewVersionStatusChecker(versionCheckURL, options.versionOverride, apiClient)
-
-			err = checkStatus(os.Stdout, kubeApi, grpcStatusChecker, versionStatusChecker)
-			if err != nil {
-				os.Exit(2)
-			}
+			configureAndRunChecks(options)
```

Comment:
I know this isn't part of the change, but the use of the possessive pronoun "your" is a little jarring especially when compared to other command descriptions. We may want to revisit this in a new PR.

Reply:
Ahhh, yeah, this is great feedback. I partially rewrote the description for this command in #1495, and I'll address this suggestion there.

As far as printing the skipped check goes, I'm a bit on the fence. If the user fails a fatal check, then they must fix it before being able to successfully use the CLI. In that case, printing the checks that were skipped might distract from the failure that requires fixing.